### PR TITLE
Option for user to create his own dashed styles and edge animations

### PR DESCRIPTION
**Pull request by janithmehta**
**Dashed Edge Styles**
    

The user can provide an array (Eg:- [6, 3, 4, 1]) or a number (Eg: 6) for the dashed styles. 

**Dash Offset**
The user has to provide an offset (Eg: 2, -2)

**These work only when the the edge-style is set to 'dashed'**

Ref: #2160

### CODE REVIEW

File: src/style/properties.js [hunk 1]
```diff
@@ -285,6 +287,8 @@ const styfn = {};
     { name: 'line-color', type: t.color },
     { name: 'line-fill', type: t.fill },
     { name: 'line-cap', type: t.lineCap },
+    { name: 'dash-style', type: t.dashStyle },
```
[the remark this answers is not in the document]
Comment (maxkfranz) [reply src/style/properties.js]: These properties should all be prefixed by `line-` to be consistent with the existing line properties.  

If we want a global `dash-style`, clients may expect it to work on things like node borders.  It's probably best to keep things separate.

File: src/style/properties.js [hunk 2]
```diff
@@ -35,6 +35,8 @@ const styfn = {};
     numbers: { number: true, unitless: true, multiple: true },
     positiveNumber: { number: true, unitless: true, min: 0, strictMin: true },
     size: { number: true, min: 0 },
+    dashStyle: { number: true, min: 0, multiple: true },
+    lineDashOffset: { number: true },
```
Comment (maxkfranz) [inline src/style/properties.js]: I don't think these types are necessary.  `number` and `numbers` should suffice.    Future rendering implementations might not allow for with-units values.  And I remember some browsers in the past not yielding precise results, so it's probably best to just consider the values to be unitless and relative.

**Comment by janithmehta** [comment]
@maxkfranz Made the required changes

**Comment by maxkfranz** [comment]
I tried this out with a marching ants animation: 

marching-ants

It seems to work well, though the gif isn't as smooth as the real thing.

**Comment by RaheelaKhan1172** [comment]
@maxkfranz @janithmehta Other than supplying `line-dash-offset` and `line-dash-pattern`, are there other steps/options necessary in order to get the marching ant animation to work?

**Pull request merged**